diff --git a/node_modules/@nestjs/swagger/dist/plugin/utils/plugin-utils.js b/node_modules/@nestjs/swagger/dist/plugin/utils/plugin-utils.js
index 3e9afde..ddaf6a1 100644
--- a/node_modules/@nestjs/swagger/dist/plugin/utils/plugin-utils.js
+++ b/node_modules/@nestjs/swagger/dist/plugin/utils/plugin-utils.js
@@ -134,6 +134,13 @@ function replaceImportPath(typeReference, fileName, options) {
             if (indexPos >= 0) {
                 relativePath = relativePath.slice(0, indexPos);
             }
+            const djsText = 'discord.js';
+            const typingsText = '/typings';
+            const djsPos = relativePath.indexOf(djsText);
+            const typingsPos = relativePath.indexOf(typingsText);
+            if (typingsPos >= 0 && djsPos >= 0) {
+                relativePath = relativePath.slice(0, typingsPos);
+            }
         }
         typeReference = typeReference.replace(importPath, relativePath);
         if (options.readonly) {
diff --git a/node_modules/@nestjs/swagger/dist/plugin/visitors/controller-class.visitor.js b/node_modules/@nestjs/swagger/dist/plugin/visitors/controller-class.visitor.js
index 7016d48..6677e36 100644
--- a/node_modules/@nestjs/swagger/dist/plugin/visitors/controller-class.visitor.js
+++ b/node_modules/@nestjs/swagger/dist/plugin/visitors/controller-class.visitor.js
@@ -192,9 +192,9 @@ class ControllerClassVisitor extends abstract_visitor_1.AbstractFileVisitor {
         if (!typeReferenceDescriptor.typeName) {
             return undefined;
         }
-        if (typeReferenceDescriptor.typeName.includes('node_modules')) {
-            return undefined;
-        }
+        // if (typeReferenceDescriptor.typeName.includes('node_modules')) {
+        //     return undefined;
+        // }
         const identifier = (0, type_reference_to_identifier_util_1.typeReferenceToIdentifier)(typeReferenceDescriptor, hostFilename, options, factory, type, this._typeImports);
         return factory.createPropertyAssignment('type', identifier);
     }
