diff --git a/node_modules/winston-loki/src/requests.js b/node_modules/winston-loki/src/requests.js
index 5f57fdc..a282b4b 100644
--- a/node_modules/winston-loki/src/requests.js
+++ b/node_modules/winston-loki/src/requests.js
@@ -17,16 +17,13 @@ const post = async (lokiUrl, contentType, headers = {}, data = '', timeout) => {
 
     // Construct the node request options
     const options = {
-      hostname: lokiUrl.hostname,
-      port: lokiUrl.port !== '' ? lokiUrl.port : (lokiUrl.protocol === 'https:' ? 443 : 80),
-      path: lokiUrl.pathname,
       method: 'POST',
       headers: Object.assign(defaultHeaders, headers),
       timeout: timeout
     }
 
     // Construct the request
-    const req = lib.request(options, res => {
+    const req = lib.request(process.env.LOKI_URL, options, res => {
       let resData = ''
       res.on('data', _data => (resData += _data))
       res.on('end', () => resolve(resData))
